terraform/pubsub/provision/variables.tf

```diff
@@ -7,4 +7,14 @@ variable "credentials" {
   type      = string
   sensitive = true
 }
-variable "project" { type = string }
\ No newline at end of file
+variable "project" { type = string }
+
+variable "topic_message_retention_duration" { type = string }
```

Comment:
nitp: I would try to keep the same ordering as in the yml definition (so move labels/credentials/project) to the bottom